server/auth_utils.go

```diff
@@ -0,0 +1,152 @@
+/*******************************************************************************
```

Comment:
Shouldn't have generic names like utils for the file. This seems to be mainly the sudo code, so call it sudo.go? Or have all the user-related code in here and call it user.go?